src/kudu/util/hdr_histogram.cc

```diff
@@ -324,6 +325,15 @@ uint64_t HdrHistogram::ValueAtPercentile(double percentile) const {
   return 0;
 }
 
+void HdrHistogram::resetHistogram(){
+  std::lock_guard<rw_spinlock> lock(histogram_mutex_);
```

Comment:
So, every getMetric() call in the plugin will end up taking as many locks as the number of histograms created in the system. If there are 100 metrics, does this mean we are taking 100 locks every few seconds when metics is collected? Wonder if this can be done in a lock free manner.

Reply:
In the regular update path we are taking the lock in shared_mode which should be quick, so hot update path should not be affected. I expect the these locks to be live for very small duration and thus multiple threads waiting on refreshing and updating same histogram should be very unlikely.
I checked the reset in java [ConcurrentHDRHistogram](https://github.com/HdrHistogram/HdrHistogram/blob/master/src/main/java/org/HdrHistogram/ConcurrentHistogram.java#L437) and they also cleared the counters under a lock. 

If you have any better ideas, let me know.